segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -263,60 +375,107 @@ public void commitRecords(Collection<SystemJournalRecord> records) throws ChunkS
+            }
 
-        // Find latest epoch with snapshot.
-        for (epochToCheck = epoch - 1; epochToCheck >= 0; epochToCheck--) {
-            snapshotFile = getSystemJournalChunkName(containerId, epochToCheck, 0);
-            if (chunkStorage.exists(snapshotFile).get()) {
-                SystemSnapshotRecord systemSnapshot = null;
-                try {
-                    // Read contents.
-                    byte[] contents = getContents(snapshotFile);
-                    systemSnapshot = SYSTEM_SNAPSHOT_SERIALIZER.deserialize(contents);
-                } catch (EOFException e) {
-                    log.warn("SystemJournal[{}] Incomplete snapshot found, skipping {}.", containerId, snapshotFile);
-                }
-                if (null != systemSnapshot) {
-                    log.debug("SystemJournal[{}] Processing system log snapshot {}.", containerId, systemSnapshot);
-                    // Initialize the segments and their chunks.
-                    for (SegmentSnapshotRecord segmentSnapshot : systemSnapshot.segmentSnapshotRecords) {
-                        // Update segment data.
-                        segmentSnapshot.segmentMetadata.setActive(true)
-                                .setOwnershipChanged(true)
-                                .setStorageSystemSegment(true);
-                        segmentSnapshot.segmentMetadata.setOwnerEpoch(epoch);
-
-                        // Add segment data.
-                        txn.create(segmentSnapshot.segmentMetadata);
-
-                        // make sure that the record is marked pinned.
-                        txn.markPinned(segmentSnapshot.segmentMetadata);
-
-                        // Add chunk metadata and keep track of start offsets for each chunk.
-                        long offset = segmentSnapshot.segmentMetadata.getFirstChunkStartOffset();
-                        for (ChunkMetadata metadata : segmentSnapshot.chunkMetadataCollection) {
-                            txn.create(metadata);
-
-                            // make sure that the record is marked pinned.
-                            txn.markPinned(metadata);
-
-                            chunkStartOffsets.put(metadata.getName(), offset);
-                            offset += metadata.getLength();
-                        }
-                        found = true;
-                    }
-                    break;
+            // Read contents.
+            try {
+                byte[] snapshotContents = getContents(snapshotFileName);
+                val systemSnapshot = SYSTEM_SNAPSHOT_SERIALIZER.deserialize(snapshotContents);
+                log.debug("SystemJournal[{}] snapshot found and parsed. {}", containerId, persisted);
+                log.debug("SystemJournal[{}] Done finding snapshots.", containerId);
+                return systemSnapshot;
+            } catch (Exception e) {
+                String object = persisted.toString();
```

Comment:
You don't need this. The logger will invoke `toString` for you

Reply:
Fixed.